packages/quantic/force-app/main/default/lwc/quanticSort/__tests__/quanticSort.test.js

```diff
@@ -0,0 +1,223 @@
+  state: mockSearchStatusState,
+  subscribe: jest.fn((callback) => {
+    callback();
+    return jest.fn();
+  }),
+};
+
+const functionsMocks = {
+  buildSort: jest.fn(() => ({
+    state: {},
+    subscribe: functionsMocks.subscribe,
+  })),
+  buildCriterionExpression: jest.fn((criterion) => criterion),
+  buildRelevanceSortCriterion: jest.fn(() => 'relevance'),
+  buildDateSortCriterion: jest.fn(() => 'date'),
+  buildSearchStatus: jest.fn(() => mockSearchStatus),
+  subscribe: jest.fn((cb) => {
+    cb();
+    return functionsMocks.unsubscribe;
+  }),
+  unsubscribe: jest.fn(() => {}),
+};
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+  variant: 'default',
+};
+
+function createTestComponent(options = defaultOptions) {
+  prepareHeadlessState();
+
+  const element = createElement('c-quantic-sort', {
+    is: QuanticSort,
+  });
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+  document.body.appendChild(element);
+  return element;
+}
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildSort: functionsMocks.buildSort,
+      buildCriterionExpression: functionsMocks.buildCriterionExpression,
+      buildSearchStatus: functionsMocks.buildSearchStatus,
+      buildRelevanceSortCriterion: functionsMocks.buildRelevanceSortCriterion,
+      buildDateSortCriterion: functionsMocks.buildDateSortCriterion,
+      SortOrder: {
+        Descending: 'descending',
+        Ascending: 'ascending',
+      },
+    };
+  };
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  // eslint-disable-next-line @lwc/lwc/no-async-operation
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticSort && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function cleanup() {
+  // The jsdom instance is shared across test cases in a single file so reset the DOM
+  while (document.body.firstChild) {
+    document.body.removeChild(document.body.firstChild);
+  }
+  jest.clearAllMocks();
+  isInitialized = false;
+}
+
+describe('c-quantic-sort', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    mockBueno();
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('construct itself without throwing', () => {
+    expect(() =>
+      createElement('c-quantic-sort', {is: QuanticSort})
+    ).not.toThrow();
+  });
+
+  describe('controller initialization', () => {
+    it('should subscribe to the headless state changes', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.subscribe).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  sortVariants.forEach((variant) => {
```

Comment:
To handle these test situations, usually there is two choices:

1. use a forEach to avoid re-writing the same similar code.
2. do not use a forEach  and re-write code but win in clarity.

In this test, I see that we are using a forEach and rewriting code because we are treating each situation separately by using `ifs`.
so there is something to improve here.

You either go with 1, in that case you need to improve your `sortVariants` object by having in it all the data you need to test the two cases without `ifs`.

Or you go with 2, and you opt for clarity and have two different `describe` to reflect the two variants.